api/v1/policy_webhook.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package v1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var policylog = logf.Log.WithName("policy-resource")
+
+func (r *Policy) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-policy-open-cluster-management-io-v1-policy,mutating=false,failurePolicy=fail,sideEffects=None,groups=policy.open-cluster-management.io,resources=policies,verbs=create;update,versions=v1,name=policy.open-cluster-management.io.webhook,admissionReviewVersions=v1
```

Comment:
Do we actually want `failurePolicy=fail`? If the webhook is down, it would allow the invalid policy to be created but the Propagator eventually reports a failure so I think I'd rather be conservative here to not cause an outage.

Reply:
I'm saying that I would prefer the policy to still be created if the webhook is down since nothing damaging actually happens, it's just you get an error once the policy propagator has processed the policy.